executor/src/core/Perspective.ts

```diff
@@ -509,14 +515,11 @@ export default class Perspective {
 
         const perspective = this.plain();
         this.#prologNeedsRebuild = true;
-        this.#pubsub.publish(PubSub.LINK_REMOVED_TOPIC, {
-            perspective: perspective,
-            link: oldLink
-        })
-        this.#pubsub.publish(PubSub.LINK_ADDED_TOPIC, {
-            perspective: perspective,
-            link: newLinkExpression
-        })
```

Comment:
Could it be that tests are failing because of these removals? Should we not trigger all three signals?

Reply:
Fixed this it was just an error due to an undefined callback array 😅